### PR TITLE
Bug 2039321: SDN: Expose controller metrics for collection

### PR DESCRIPTION
**Pull request by martinkennelly**
Signed-off-by: Martin Kennelly <mkennell@redhat.com>

/cc

Depends on: https://github.com/openshift/sdn/pull/388

### CODE REVIEW

**Comment by martinkennelly** [comment]
verify test is broken due to a dependency repository getting nuked.

An issue I noticed with this PR is that only one of the SDN controller endpoints is active and exposes a HTTP endpoint. The inactive controller rbac proxy returns 504 bad gateway back to the caller. This isn't a problem but it would be nice not to waste this CPU time polling dead endpoints. If anyone has any ideas to improve this situation..

**Comment by martinkennelly** [comment]
/cc @danwinship This PR is to expose SDN metrics from PR: https://github.com/openshift/sdn/pull/358 
Could you review since you have context?

**Comment by martinkennelly** [comment]
rebased due to conflict

**Review by alexanderConstantinescu — APPROVED** [review]
/lgtm

**Comment by martinkennelly** [comment]
/retest-required
Issues not related to this patch.
Infra / terraform failure.

**Comment by martinkennelly** [comment]
Retesting after a merge of dependencies and associated image should now be available.

**Comment by martinkennelly** [comment]
Job e2e-agnostic-upgrade failed with a known issue: https://bugzilla.redhat.com/show_bug.cgi?id=1972948

File: bindata/network/openshift-sdn/controller.yaml [hunk 1]
```diff
@@ -54,6 +54,59 @@ spec:
...
+            # As the secret mount is optional we must wait for the files to be present.
+            # The service is created in monitor.yaml and this is created in controller.yaml.
+            # If it isn't created there is probably an issue so we want to crashloop.
+            TS=$(date +%s)
+            WARN_TS=$(( ${TS} + $(( 20 * 60)) ))
+            HAS_LOGGED_INFO=0
+
+            log_missing_certs(){
+                CUR_TS=$(date +%s)
+                if [[ "${CUR_TS}" -gt "WARN_TS"  ]]; then
```
Comment (ricky-rav) [inline bindata/network/openshift-sdn/controller.yaml:77]: Here you're probably missing a dollar sign in front of `WARN_TS`, right? :)

Comment (martinkennelly) [reply bindata/network/openshift-sdn/controller.yaml:77]: Yep. Thank you!

**Pull request merged**